### DISCUSSION TITLE
Async DoneFunction in QSelect->onFilter->DoneFucntion

### DISCUSSION THREAD

**Discussion by gnubyte** [quasarframework/quasar#12407] [Ideas / Proposals] [answered]
Can we get an async of the done function in the QSelect?

The ability to await a `fetch` call to an API is desirable here.

![image](https://user-images.githubusercontent.com/16541000/153095876-1217a710-c0b5-482b-9b64-57f5a302dfbd.png)

Essentially I want to use the done function, to run a live search to an API, return the search results, then populate the dropdowns new options from my API. 

https://v1.quasar.dev/vue-components/select#the-options
https://v1.quasar.dev/vue-components/select#qselect-api

**Comment by metalsadman** [comment, marked as answer]
do your api call before it, meaning you can make your filter function async, await your api call inside, then call your doneFn when you receive the result.
see this example that mimics it https://quasar.dev/vue-components/select#example--lazy-filtering or in v1 https://v1.quasar.dev/vue-components/select#example--lazy-filtering
